src/main/java/com/hubspot/imap/client/ImapClient.java

```diff
@@ -199,6 +205,19 @@ private void stopKeepAlive() {
     return send(new OpenCommand(folderName, openMode));
   }
 
+  public CompletableFuture<TaggedResponse> append(String folderName, Set<MessageFlag> flags, Optional<ZonedDateTime> dateTime, ImapMessage message) throws Exception {
+    StringLiteralCommand stringLiteralCommand = new StringLiteralCommand(message.bodyToString());
+    AppendCommand appendCommand = new AppendCommand(folderName, flags, dateTime, stringLiteralCommand.size(message.getBody().getCharset()));
+
+    return CompletableFutures.handleCompose(send(appendCommand), (BiFunction<ImapResponse, Throwable, CompletableFuture<TaggedResponse>>) (imapResponse, throwable) -> {
```

Comment:
I am not sure I love having logic like this live in the `ImapClient` itself. I just feel like this class is going to grow to be very large/complex if we allow this. This may be necessary as is, but I really want a way to separate out response handling logic...

Reply:
Yeah, I get that. I could pull the bifunction out of there and define a `HandleAppendResponseFunction` class in its own file to avoid bloating the `ImapClient`. We could have a `functions` package and pull out the functions in `login(...)` and `startTls()` that follow a similar pattern. Only thing is that I would have to instantiate the functions with the `ImapClient` itself in order to do the following sends which feels a little weird. i.e 
```java
return CompletableFutures.handleCompose(send(appendCommand), new HandleAppendResponseFunction(this, appendCommand)).toCompletableFuture();
```

What do you think?

Reply:
hrrm I wonder if we could just make this a method of the command itself. This whole get-continuation-then-repsond pattern is fairly common, so why not have a `ContinuableCommand` which itself has a `continue(ImapResponse)` method?

Reply:
We would still have to instantiate the command with the `ImapClient` in order to do the subsequent send. So the `AppendCommand` would extend the `ContinuableCommand` which essentially just specifies that it has an `ImapClient` and a `continueAfterResponse` method:

```java
AppendCommand appendCommand = new AppendCommand(this, folderName, flags, dateTime, message);
return CompletableFutures.handleCompose(send(appendCommand), appendCommand::continueAfterResponse).toCompletableFuture();
```

Which still feels a little weird to me. Is this what you were thinking?